packages/react/src/components/DataTable/TableToolbarSearch.js

```diff
@@ -51,6 +51,18 @@ const TableToolbarSearch = ({
     }
   }, [controlled, expandedState]);
 
+  //eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(
+    () => {
+      if (defaultValue) {
+        setExpandedState(true);
+        onChangeProp('', defaultValue);
```

Comment:
Do we need to call `onChange`  if they've provided us the default value?

Reply:
Without this call, the table was not automatically filtering the values on load.

Reply:
Ah, got it. Given that this is in `useEffect` do we want the flicker while rendering the filtered list? Seems like we'd need a way to hoist up this default value so that this line is initialized appropriately, right? https://github.com/carbon-design-system/carbon/blob/dc860ba78dbac515a6910d05ed0e34fb941be797/packages/react/src/components/DataTable/state/getDerivedStateFromProps.js#L33

Reply:
Hmm, not noticing any flicker on my end and its rendering properly... 🤔  also unsure about how to hoist that value up

Reply:
One option could be adding `defaultSearchValue` or something to DataTable and have it propagate down through an `inputValue` render prop but I totally get not wanting to go down that route.

The only reason I think it came up is that it seems like the render order becomes

- render `DataTable` with search value `''`
- render `TableToolbarSearch` with defaultValue `'foo'`
- run effect in `TableToolbarSearch` and update `DataTable` input value with `'foo'`
- re-render `DataTable` (which filters rows)

Just was curious if we could have this happen in one phase instead of two.

Reply:
I'm thinking we could get it into 1 phase as well, but I think I'd like to split that into a separate issue and get this functionality in to fix the issue. That work?

Reply:
For sure! The only concern I would have would be if the API for the components change as a result. It seemed (initially) that in order to get it all done in one render that we'd need to supply a prop to DataTable and then have it pass things down to `TableToolbarSearch`.

If we think the API will change as a result, it seems like it'd make sense to get it done in one go instead of potentially adding and then removing props like `defaultValue` if they're not relevant anymore.

Reply:
Ah so just remove `defaultValue` from `TableToolbarSearch` altogether, and pass it down from `DataTable` instead?

Reply:
Honestly I'm curious what you think would make most sense here, just an idea. We could totally do something like:

```jsx
<DataTable defaultInputValue="foo">
  {({ onInputChange, inputValue }) => (
    <TableToolbarSearch onChange={onInputChange} value={inputValue} />
  )}
</DataTable>
```

But definitely haven't spent as much time in this part of the codebase recently so might not make sense? Let me know!

Reply:
Tried a couple of different ways of implementing it, but personally, it seems more confusing than just letting users put a `defaultValue` on the `TableToolbarSearch` and it just works  🤷‍♂